blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/GaugeRollupSerializerTest.java

```diff
@@ -0,0 +1,75 @@
+
+    @Test
+    public void testSerializerDeserializerV1() throws Exception {
+        BluefloodGaugeRollup gauge1 = BluefloodGaugeRollup.buildFromRawSamples(
+                                        asPoints(SimpleNumber.class, System.currentTimeMillis(), 300, new SimpleNumber(10L)));
+        BluefloodGaugeRollup gauge2 = BluefloodGaugeRollup.buildFromRawSamples(
+                asPoints(SimpleNumber.class, System.currentTimeMillis()-100, 300, new SimpleNumber(1234567L)));
+        BluefloodGaugeRollup gauge3 = BluefloodGaugeRollup.buildFromRawSamples(
+                asPoints(SimpleNumber.class, System.currentTimeMillis()-200, 300, new SimpleNumber(10.4D)));
+        BluefloodGaugeRollup gaugesRollup = BluefloodGaugeRollup.buildFromGaugeRollups(
+                asPoints(BluefloodGaugeRollup.class, System.currentTimeMillis(), 300, gauge1, gauge2, gauge3));
+        Assert.assertEquals(3, gaugesRollup.getCount());
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        baos.write(Base64.encodeBase64(new NumericSerializer.GaugeRollupSerializer().toByteBuffer(gauge1).array()));
+        baos.write("\n".getBytes());
+        baos.write(Base64.encodeBase64(new NumericSerializer.GaugeRollupSerializer().toByteBuffer(gauge2).array()));
+        baos.write("\n".getBytes());
+        baos.write(Base64.encodeBase64(new NumericSerializer.GaugeRollupSerializer().toByteBuffer(gauge3).array()));
+        baos.write("\n".getBytes());
+        baos.write(Base64.encodeBase64(new NumericSerializer.GaugeRollupSerializer().toByteBuffer(gaugesRollup).array()));
+        baos.write("\n".getBytes());
+        baos.close();
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(baos.toByteArray())));
+
+        ByteBuffer bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodGaugeRollup deserializedGauge1 = NumericSerializer.serializerFor(BluefloodGaugeRollup.class).fromByteBuffer(bb);
+        Assert.assertEquals(gauge1, deserializedGauge1);
+
+        bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodGaugeRollup deserializedGauge2 = NumericSerializer.serializerFor(BluefloodGaugeRollup.class).fromByteBuffer(bb);
+        Assert.assertEquals(gauge2, deserializedGauge2);
+
+        bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodGaugeRollup deserializedGauge3 = NumericSerializer.serializerFor(BluefloodGaugeRollup.class).fromByteBuffer(bb);
+        Assert.assertEquals(gauge3, deserializedGauge3);
+
+        bb = ByteBuffer.wrap(Base64.decodeBase64(reader.readLine().getBytes()));
+        BluefloodGaugeRollup deserializedGauge4 = NumericSerializer.serializerFor(BluefloodGaugeRollup.class).fromByteBuffer(bb);
+        Assert.assertEquals(gaugesRollup, deserializedGauge4);
+
+        Assert.assertFalse(deserializedGauge1.equals(deserializedGauge2));
```

Comment:
Some people might split this into 3 tests, I think @izrik prefer things that way.  We can cover this in our dev meeting.